core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -250,6 +250,9 @@ public void run() {
       loadBalancer.shutdown();
       loadBalancer = null;
```

Comment:
A factory that recycles the instances is violating the API. If the factory is shared among multiple channels, worse things will happen anyway. I don't think we should worry about it.